src/protocols/dhcpv4/encode.c

```diff
@@ -70,13 +66,10 @@ static inline VALUE_PAIR *next_encodable(fr_cursor_t *cursor, void *encoder_ctx)
  */
 static inline VALUE_PAIR *first_encodable(fr_cursor_t *cursor, void *encoder_ctx)
 {
-	VALUE_PAIR	*vp;
+	VALUE_PAIR	*vp = fr_cursor_current(cursor);
 	fr_dhcpv4_ctx_t	*packet_ctx = encoder_ctx;
 
-	vp = fr_cursor_current(cursor);
-	if (is_encodable(packet_ctx->root, vp)) return vp;
-
-	return next_encodable(cursor, encoder_ctx);
+	return is_encodable(packet_ctx->root, vp) ? vp : next_encodable(cursor, encoder_ctx);
```

Comment:
We try to avoid ` ... ? ... : ...`, except in function arguments.  It makes things difficult to read.

Reply:
OK. Thanks to you and mcnewton; these should ultimately go into the coding standards.